Core/src/Detector/CylindricalDetectorHelper.cpp

```diff
@@ -0,0 +1,1115 @@
+      transform, std::move(bounds));
+
+  // A make a portal and indicate the new link direction
+  const auto& stitchBoundaries =
+      (stitchValue == Acts::binZ) ? zBoundaries : phiBoundaries;
+  return Acts::Experimental::PortalReplacement(
+      Acts::Experimental::Portal::makeShared(surface), index, nDir,
+      stitchBoundaries, stitchValue);
+}
+
+/// @brief Helper function to create a disc portal replacement
+/// @param gcxt the geometry context of thix call
+/// @param volumeCenter a reference center of the volume (combined)
+/// @param refSurface the reference surface (old portal)
+/// @param boundaries the vector of binning boundaries in r
+/// @param binning the binning of the sector (inR, inZ)
+/// @param index the index of this portal to be set
+/// @param nDir the navigation direction to be set
+///
+/// @return a new portal replacement object
+Acts::Experimental::PortalReplacement createSectorReplacement(
+    const Acts::GeometryContext& gctx, const Acts::Vector3& volumeCenter,
+    const Acts::Surface& refSurface,
+    const std::vector<Acts::ActsScalar>& boundaries, Acts::BinningValue binning,
+    unsigned int index, Acts::NavigationDirection nDir) {
+  // Get a reference transform
+  const auto& refTransform = refSurface.transform(gctx);
+  auto refRotation = refTransform.rotation();
+  // Bounds handling
+  const auto& boundValues = refSurface.bounds().values();
+  std::unique_ptr<Acts::PlanarBounds> bounds = nullptr;
+
+  // Create a new transform
+  Acts::Transform3 transform = Acts::Transform3::Identity();
+  if (binning == Acts::binR) {
+    // Center r calculation
+    Acts::ActsScalar centerR =
+        0.5 * (boundaries[0u] + boundaries[boundaries.size() - 1u]);
+    // New joint center
+    Acts::Vector3 pCenter = volumeCenter + centerR * refRotation.col(2);
+    transform.pretranslate(pCenter);
+    Acts::ActsScalar halfX =
+        0.5 * (boundValues[Acts::RectangleBounds::BoundValues::eMaxX] -
+               boundValues[Acts::RectangleBounds::BoundValues::eMinX]);
+    // New joint bounds
+    bounds = std::make_unique<Acts::RectangleBounds>(
+        halfX, 0.5 * (boundaries[boundaries.size() - 1u] - boundaries[0u]));
+  } else if (binning == Acts::binZ) {
+    // Center R calculation, using projection onto vector
+    const auto& surfaceCenter = refSurface.center(gctx);
+    Acts::Vector3 centerDiffs = (surfaceCenter - volumeCenter);
+    Acts::ActsScalar centerR = centerDiffs.dot(refRotation.col(2));
+    // New joint center
+    Acts::Vector3 pCenter = volumeCenter + centerR * refRotation.col(2);
+    transform.pretranslate(pCenter);
+    // New joint bounds
+    Acts::ActsScalar halfY =
+        0.5 * (boundValues[Acts::RectangleBounds::BoundValues::eMaxY] -
+               boundValues[Acts::RectangleBounds::BoundValues::eMinY]);
+    bounds = std::make_unique<Acts::RectangleBounds>(
+        0.5 * (boundaries[boundaries.size() - 1u] - boundaries[0u]), halfY);
```

Comment:
could we use `Acts::range_medium(boundaries)` here?

Reply:
not on the boundValues, yes on the boundaries - see above.